### PR TITLE
Tighten down healthcheck timings, use `autoheal` to do the restarting

### PR DESCRIPTION
**Pull request by staticfloat**
This should make it easier for us to debug healthcheck issues

### CODE REVIEW

**Comment by staticfloat** [comment]
@fredrikekre my current theory why there was so much weirdness with the healthcheck stuff is that `curl` ended up getting slowed down by things, and we ended up piling on many different processes.  I also didn't like the solution of running `kill` inside of that healthcheck, (especially since we had multiple layers of retries in there) so I decided to simplify the `healthcheck` stuff to not restart the container, but instead use another container that just searches for unhealthy services and restarts them from outside of the container.

**Comment by staticfloat** [comment]
This is already deployed to `eu-central2`, so we can see if it works as expected there.

**Comment by fredrikekre** [comment]
It has been unhealthy for 20 minutes now without any restart. https://github.com/willfarrell/docker-autoheal#env-defaults says it should check every 5s so something is wrong. Also, something seems to be very broken with this particular VM, why does it go unhealthy almost directly?

**Review by fredrikekre — APPROVED** [review]

File: deployment/docker-compose.yml [hunk 1]
```diff
@@ -19,12 +19,13 @@ services:
             JULIA_PKG_SERVER_STORAGE_SERVERS: "${STORAGE_SERVERS:-https://us-east.storage.juliahub.com,https://kr.storage.juliahub.com}"
         labels:
             com.centurylinklabs.watchtower.scope: "pkgserver.jl"
+            autoheal: "true"
         healthcheck:
-            test: curl -f --retry 5 --max-time 5 --retry-delay 10 --retry-max-time 60 "http://localhost:8000/meta" || bash -c 'kill -s 15 $$(pidof julia) && (sleep 10; kill -s 9 $$(pidof julia))'
-            interval: 2m
-            timeout: 10s
+            test: ["CMD", "curl", "-f", "-m", "1", "http://localhost:8000/meta"]
```
Comment (fredrikekre) [inline deployment/docker-compose.yml:24]: 1 sec is ok?

Comment (staticfloat) [reply deployment/docker-compose.yml:24]: It usually completes in <20ms for me, so I think 1s should be plenty.  In any case, it should take 3 failures in a row to cause it to get killed.

**Pull request merged**